google/cloud/storage/examples/storage_object_samples.cc

```diff
@@ -35,84 +33,117 @@ char const* ConsumeArg(int& argc, char* argv[]) {
   return result;
 }
 
+std::string command_usage;
+
 void PrintUsage(int argc, char* argv[], std::string const& msg) {
   std::string const cmd = argv[0];
   auto last_slash = std::string(cmd).find_last_of('/');
   auto program = cmd.substr(last_slash + 1);
   std::cerr << msg << "\nUsage: " << program << " <command> [arguments]\n\n"
-            << "Examples:\n";
-  for (auto const& example :
-       {"read-object <bucket-name> <object-name>",
-        "insert-object <object name> <object contents>"}) {
-    std::cerr << "  " << program << " " << example << "\n";
+            << "Commands:\n"
+            << command_usage << std::endl;
+}
+
+void ListObjects(google::cloud::storage::Client client, int& argc,
+                 char* argv[]) {
+  if (argc < 2) {
+    throw Usage{"list-objects <bucket-name>"};
+  }
+  auto bucket_name = ConsumeArg(argc, argv);
+  //! [list objects] [START storage_list_files]
+  namespace gcs = google::cloud::storage;
+  [](gcs::Client client, std::string bucket_name) {
+    for (auto const& meta : client.ListObjects(bucket_name)) {
+      std::cout << "bucket_name=" << meta.bucket()
+                << ", object_name=" << meta.name() << std::endl;
+    }
   }
-  std::cerr << std::flush;
+  //! [list objects] [END storage_list_files]
+  (std::move(client), bucket_name);
 }
 
-//! [insert object]
-void InsertObject(gcs::Client client, int& argc, char* argv[]) {
+void InsertObject(google::cloud::storage::Client client, int& argc,
+                  char* argv[]) {
   if (argc < 3) {
     throw Usage{
         "insert-object <bucket-name> <object-name> <object-contents (string)>"};
   }
   auto bucket_name = ConsumeArg(argc, argv);
   auto object_name = ConsumeArg(argc, argv);
   auto contents = ConsumeArg(argc, argv);
-  auto meta =
-      client.InsertObject(bucket_name, object_name, std::move(contents));
-  std::cout << "The new object metadata is " << meta << std::endl;
+  //! [insert object] [START createBlobFromByteArray]
```

Comment:
Remove tag for now we don't have this sample in cloud documentation.

Reply:
Will do, but I got it from here:

https://github.com/GoogleCloudPlatform/google-cloud-java/blob/9e320b3a29792cd87b5ede7893af42457b56404c/google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/BucketSnippets.java#L197

Linked from here:

https://cloud.google.com/storage/docs/uploading-objects#storage-upload-object-java

Reply:
Ah. that was my mistake, sorry. 
Good catch. Instead of removing it, please update region tag to `upload_file`.

Reply:
Done.  But just in case, did you mean `storage_upload_file`?

Reply:
Ah you're right, it's `storage_upload_file`. I'll add an internal link in meeting notes to where all the canonical region tags are defined.

Reply:
LGTM after that. sorry!

Reply:
Fixed.